src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1052,33 +1058,24 @@ ucs_status_t uct_ud_ep_flush_nolock(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
     }
 
     /* Expedite acknowledgment on the last skb in the window */
-    if (uct_ud_ep_is_last_ack_received(ep)) {
+    if (uct_ud_ep_is_last_ack_received(ep) &&
+        !(ep->flags & UCT_UD_EP_FLAG_FLUSH_CANCEL)) {
```

Comment:
IMO better pass flags to that function and check "flags & UCT_FLUSH_FLAG_CANCEL" - instead of checking ep->flags